source/base/Text.ooc

```diff
@@ -0,0 +1,298 @@
+		this split(This new(separator&, 1, Owner Stack))
+	}
+	split: func ~withString (separator: String) -> VectorList<This> {
+		this split(This new(separator))
+	}
+	split: func ~withText (separator: This) -> VectorList<This> {
+		result := VectorList<This> new()
+		start := 0
+		pos := this find(separator)
+		while (pos != -1) {
+			if (pos > start)
+				result add(this slice(start, pos - start))
+			start = pos + separator count
+			pos = this find(separator, start)
+		}
+		if (start < this count && pos == -1)
+			result add(this slice(start, this count - start))
+		result
+	}
+	toString: func -> String {
+		String new(this _buffer, this count)
+	}
+	toInt: func -> Int {
+		this toLLong() as Int
+	}
+	toLong: func -> Long {
+		this toLLong() as Long
+	}
+	toLLong: func -> LLong {
+		this toLLong~inBase(this _detectNumericBase())
+	}
+	toInt: func ~inBase (base: Int) -> Int {
+		this toLLong~inBase(base) as Int
+	}
+	toLong: func ~inBase (base: Int) -> LLong {
+		this toLLong~inBase(base) as Long
+	}
+	toLLong: func ~inBase (base: Int) -> LLong {
+		result := 0 as LLong
+		if (this isEmpty == false) {
+			if (this[0] == '-')
+				result = -1 * this slice(1, this count - 1) toULong~inBase(base)
+			else
+				result = this toULong~inBase(base) as LLong
+		}
+		result
+	}
+	toULong: func -> ULong {
+		this toULong~inBase(this _detectNumericBase())
+	}
+	toULong: func ~inBase (base: Int) -> ULong {
+		result := 0 as ULong
+		if (this isEmpty == false) {
+			lastValidIndex := -1
+			start := 0
+			if (base == 16 && this beginsWith(This new(c"0x", 2))) {
+				start += 2
+				lastValidIndex += 2
+			}
+			for (i in start .. this count) {
+				if (This _isNumeric(this[i], base))
+					++lastValidIndex
+				else
+					break
+			}
+			power := 1 as ULong
+			while (lastValidIndex >= 0) {
+				result += power * This _toInt(this[lastValidIndex], base)
+				--lastValidIndex
+				power *= base
+			}
+		}
+		result
+	}
+	toFloat: func -> Float {
+		this toLDouble() as Float
+	}
+	toDouble: func -> Double {
+		this toLDouble() as Double
+	}
+	toLDouble: func -> LDouble {
+		result := 0.0 as LDouble
+		if (!this isEmpty) {
+			sign := 1
+			start := 0
+			if (this[0] == '-') {
+				start = 1
+				sign = -1
+			}
+			index := this count
+			for (i in start .. this count) {
+				if (!This _isNumeric(this[i], 10)) {
+					index = i
+					break
+				}
+			}
+			result = this slice(start, index - start) toLLong~inBase(10)
+			if (index > -1 && index < this count) {
+				if (this[index] == '.') {
+					power := 0.1
+					for (i in index + 1 .. this count) {
+						if (This _isNumeric(this[i], 10)) {
+							result += power * This _toInt(this[i], 10)
+							power /= 10
+						} else {
+							index = i
+							break
+						}
+					}
+				}
+				if (this[index] == 'e' || this[index] == 'E') {
+					exponent := this slice(index + 1, this count - index) toInt~inBase(10)
+					result = result * pow(10, exponent)
+				}
+			}
+			result *= sign
+		}
+		result
```

Comment:
This is a pretty long function. Is there any way you could split it up somewhat?